src/software/gui/full_system/widgets/ai_control.cpp

```diff
@@ -163,3 +166,96 @@ void setupPlayOverrideComboBox(Ui::AutogeneratedFullSystemMainWidget *widget,
     QWidget::connect(widget->play_override_combo_box, &QComboBox::currentTextChanged,
                      on_play_changed);
 }
+
+void setupChannelSpinBox(Ui::AutogeneratedFullSystemMainWidget *widget,
+                         std::shared_ptr<ThunderbotsConfig> config)
+{
```

Comment:
can we refactor these setup__SpinBox into a generic interface like we do for slider?

Reply:
Yup I noticed that do when doing this. Made #1587 to deal with this separately.